zomes/countersigning/coordinator/src/lib.rs

```diff
@@ -0,0 +1,186 @@
+    fns.insert((zome_info()?.name, "recv_remote_signal".into()));
+    let functions = GrantedFunctions::Listed(fns);
+    create_cap_grant(CapGrantEntry {
+        tag: "".into(),
+        // empty access converts to unrestricted
+        access: ().into(),
+        functions,
+    })?;
+
+    Ok(InitCallbackResult::Pass)
+}
+
+#[hdk_extern]
+fn recv_remote_signal(signal: Signals) -> ExternResult<()> {
+    // TODO we don't know who the remote signal came from?!
+    emit_signal(signal)
+}
+
+#[hdk_extern]
+fn initiator_hello() -> ExternResult<String> {
+    Ok("Hello!".to_string())
+}
+
+#[hdk_extern]
+fn participant_hello() -> ExternResult<()> {
+    create_link(
+        hash_entry(EntryTypes::ParticipantBase(ParticipantBase))?,
+        agent_info()?.agent_initial_pubkey,
+        LinkTypes::Participant,
+        (),
+    )?;
+
+    Ok(())
+}
+
+#[hdk_extern]
+fn list_participants() -> ExternResult<Vec<AgentPubKey>> {
+    let links = get_links(
+        GetLinksInputBuilder::try_new(
+            hash_entry(EntryTypes::ParticipantBase(ParticipantBase))?,
+            LinkTypes::Participant,
+        )?
+        .build(),
+    )?;
+    Ok(links
+        .into_iter()
+        .map(|link| link.target.clone().try_into().unwrap())
+        .collect())
+}
+
+#[hdk_extern]
+fn start_two_party(with_other: AgentPubKey) -> ExternResult<PreflightResponse> {
+    let my_agent_info = agent_info()?;
+
+    let entry = ImportantAgreement {
+        best_ice_cream_flavour: "strawberry".to_string(),
+    };
+
+    let entry_hash = hash_entry(EntryTypes::ImportantAgreement(entry.clone()))?;
+
+    let session_times = session_times_from_millis(2_000)?;
+    let request = PreflightRequest::try_new(
+        entry_hash,
+        vec![
+            (
+                my_agent_info.agent_initial_pubkey,
+                vec![Roles::Vendor.into()],
+            ),
+            (with_other.clone(), vec![Roles::Taster.into()]),
+        ],
+        Vec::with_capacity(0),
+        0,
+        true,
+        session_times,
+        ActionBase::Create(CreateBase::new(
+            UnitEntryTypes::ImportantAgreement.try_into()?,
+        )),
+        PreflightBytes(entry.best_ice_cream_flavour.into_bytes()),
+    )
+    .map_err(|e| {
+        wasm_error!(WasmErrorInner::Guest(format!(
+            "Failed to create countersigning request: {:?}",
+            e
+        )))
+    })?;
+
+    // Accept ours now and then Holochain should wait for the other party to join the session
+    let my_acceptance = accept_countersigning_preflight_request(request.clone())?;
+
+    let response = match &my_acceptance {
+        PreflightRequestAcceptance::Accepted(response) => response.clone(),
+        e => {
+            return Err(wasm_error!(WasmErrorInner::Guest(format!(
+                "Unexpected response: {:?}",
+                e
+            ))))
+        }
+    };
+
+    // Let the other party know about the request
+    send_remote_signal(
```

Comment:
since remote signals are unreliable, and we only have one remote target, would it make more sense for this to be a call_remote?

Reply:
This is a good point. We'll probably have to use remote signals to keep the M of N case performant but `call_remote` is a better option here.

Reply:
I'll leave the signal setup in place to be used later but switch the init/accept zome functions to use `call_remote`